storage/range_command.go

```diff
@@ -234,7 +263,7 @@ func (r *Range) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args *
 		}
 	} else {
 		// The transaction doesn't exist yet on disk; use the supplied version.
-		reply.Txn = gogoproto.Clone(args.Txn).(*proto.Transaction)
+		reply.Txn = args.Txn
```

Comment:
yes. Can you put the merge back in with a comment to reevaluate whether we need it? Just so that I can sleep at night.